apm-agent-core/src/main/java/co/elastic/apm/agent/report/IntakeV2ReportingEventHandler.java

```diff
@@ -132,77 +167,43 @@ private void writeEvent(ReportingEvent event) {
         }
     }
 
-    private void cancelTimeout() {
-        if (timeoutTask != null) {
-            timeoutTask.cancel();
-            timeoutTask = null;
-        }
-    }
-
     @Override
     @Nullable
     protected HttpURLConnection startRequest(String endpoint) throws Exception {
         HttpURLConnection connection = super.startRequest(endpoint);
         if (connection != null) {
-            if (reporter != null) {
-                timeoutTask = new FlushOnTimeoutTimerTask(reporter);
+            if (timeoutTask != null) {
```

Comment:
Can this ever be null?

Reply:
Not in practice. But `timeoutTask` is `@Nullable` as it's initialized in the `init` method.